lib/active_model/serializer/association.rb

```diff
@@ -10,24 +10,86 @@ class Serializer
     #  Association.new(:comments, { serializer: CommentSummarySerializer })
     #
     class Association < Field
+      # TECHDEBT: remove unneeded accessors
+      # DONOTMERGE: temporary while I separate out what needs to be mutated and when
+      attr_reader :parent_serializer, :parent_serializer_options, :include_slice, :reflection, :association_options
+      def initialize(*)
+        super
+        @association_options = options
+        @parent_serializer = association_options.fetch(:parent_serializer)
+        @parent_serializer_options = association_options.fetch(:parent_serializer_options)
+        @include_slice = association_options.fetch(:include_slice)
+        @reflection = association_options.fetch(:reflection)
+
+        self.options = reflection.options.dup
+        # Pass the parent's namespace onto the child serializer
+        options[:namespace] ||= parent_serializer_options[:namespace]
+      end
+
       # @return [Symbol]
       def key
         options.fetch(:key, name)
       end
 
-      # @return [ActiveModel::Serializer, nil]
-      def serializer
-        options[:serializer]
-      end
-
       # @return [Hash]
       def links
         options.fetch(:links) || {}
       end
 
       # @return [Hash, nil]
       def meta
-        options[:meta]
+        reflection.options[:association_meta]
+      end
+
+      def object
+        @object ||= reflection.value(parent_serializer, include_slice)
+      end
+      alias eval_reflection_block object
+
+      def include_data?
+        eval_reflection_block if reflection.block
+        reflection.include_data?(include_slice)
+      end
+
+      # @return [ActiveModel::Serializer, nil]
+      def serializer
+        return @serializer if defined?(@serializer)
+        if serializer_class
+          serialize_object!(object)
+        elsif !object.nil? && !object.instance_of?(Object)
+          options[:virtual_value] = object
+        end
+        @serializer = options[:serializer]
+      end
+
+      def serialize_object!(object)
```

Comment:
These methods should all be `private`so we know what can be changed later without side effects.